packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -742,178 +738,849 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return 0;
   }
 
-  void _handleMovement(
-    LogicalKeyboardKey key, {
-    required bool wordModifier,
-    required bool lineModifier,
-    required bool shift,
-  }){
-    if (wordModifier && lineModifier) {
-      // If both modifiers are down, nothing happens on any of the platforms.
+  static TextSelection _moveGivenSelectionLeft(TextSelection selection, String text) {
```

Comment:
_moveGivenSelectionRight has a nice little top-level comment, that would probably be useful here as well.